internal/testutils/testserver.go

```diff
@@ -56,9 +62,31 @@ type ServerResponses map[HTTPMethod]struct {
 }
 
 // NewHTTPTestServer creates a new HTTP test server with the specified responses for each HTTP method.
-func NewHTTPTestServer(t *testing.T, arg ServerResponses) *TestServer {
-	return &TestServer{Server: httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		if res, found := arg[req.Method]; found {
+func NewHTTPTestServer(t *testing.T, responses []ServerResponses) *TestServer {
+	testServer := &TestServer{}
+	handler := func(rw http.ResponseWriter, req *http.Request) {
+		testServer.calls++
+		// Special case: if only one response is specified, this will be reused
+		// for each and every call
+		if len(responses) == 1 {
```

Comment:
I don't like that implicit behavior but it is useful sometimes and already existing tests are relying on that

Reply:
That seems pretty strange to use though? And gives me no way to assert that a test server receives exactly one API call, rather than many...

I don't know which tests rely on this behavior but IMO it'd be better to re-write them to define their responses and expected requests explicitly. 

Awesome to see that you introduced this idea btw!